Straight_skeleton_extrusion_2/test/Straight_skeleton_extrusion_2/test_sls_extrude.cpp

```diff
@@ -0,0 +1,287 @@
+        polys.emplace_back(poly.begin(), poly.end());
+      else
+        polys.emplace_back(poly.rbegin(), poly.rend());
+    }
+  };
+
+  if(is_number_of_CC_in_input)
+  {
+    int ccb_count = 0;
+    in >> ccb_count;
+    for(int i=0; i<ccb_count && in; ++i)
+      read_polygon(i);
+  }
+  else
+  {
+    int i = 0;
+    while(in)
+      read_polygon(i++);
+  }
+
+  if(polys.empty())
+  {
+    std::cerr << "Error: empty input?" << std::endl;
+    return false;
+  }
+
+  std::cout <<"Polygon with border of size: " << polys[0].size() << std::endl;
+  if(polys.size() > 1)
+    std::cout << polys.size() - 1 << " hole(s)" << std::endl;
+
+  p = Polygon_with_holes_2(polys[0]);
+  for(std::size_t i=0; i<polys.size()-1; ++i)
+    p.add_hole(polys[i+1]);
+
+  return true;
+}
+
+template <typename K>
+bool read_input_polygon(const char* filename,
+                        CGAL::Polygon_with_holes_2<K>& p)
+{
+  std::string ext = CGAL::IO::internal::get_file_extension(filename);
+  if(ext == "dat")
+  {
+    return read_dat_polygon(filename, p);
+  }
+  else
+  {
+    std::cerr << "Error: unknown file extension: " << ext << std::endl;
+    return false;
+  }
+}
+
+template <typename K>
+bool read_segment_speeds(const char* filename,
+                         std::vector<std::vector<typename K::FT> >& weights)
+{
+  using FT = typename K::FT;
+
+  std::ifstream in(filename);
+  if(!in)
+  {
+    std::cerr << "Error: Could not read " << filename << std::endl;
+    return false;
+  }
+
+  std::vector<FT> border_weights;
+
+  std::string line;
+  while(getline(in, line))
+  {
+    if(line.empty())
+    {
+      weights.push_back(border_weights);
+      border_weights.clear();
+    }
+
+    std::istringstream iss(line);
+    double w;
+    if(iss >> w)
+      border_weights.push_back(w);
+  }
+
+  // in case the last line is not empty
+  if(!border_weights.empty())
+    weights.push_back(border_weights);
+
+  return true;
+}
+
+std::string root_name(const std::string& full_filename)
+{
+  std::string name = std::string(full_filename);
+  name = name.substr(name.find_last_of("/") + 1, name.length() - 1);
+  name = name.substr(0, name.find_last_of("."));
+  return name;
+}
+
+template <typename K>
+bool test(const char* poly_filename,
+          const char* angles_filename,
+          const typename K::FT height,
+          const typename K::FT expected_volume)
+{
+  using FT = typename K::FT;
+
+  using Polygon_with_holes_2 = CGAL::Polygon_with_holes_2<K>;
+
+  using Point_3 = typename K::Point_3;
+  using Mesh = CGAL::Surface_mesh<Point_3>;
+
+  std::cout << "\nTest:\n"
+            << "\tPolygon: " << poly_filename << "\n"
+            << "\tAngles: " << angles_filename << "\n"
+            << "\tHeight: " << height << std::endl;
+
+  Polygon_with_holes_2 pwh;
+  if(!read_input_polygon(poly_filename, pwh) || pwh.outer_boundary().is_empty())
+  {
+    std::cerr << "Error: failure during polygon read" << std::endl;
+    return false;
+  }
+
```

Comment:
When I put
```
  auto ob = pwh.outer_boundary();
  return true;
```
here I also get the assertion that the Core memory pool is not empty.  So the runtime error in this [testsuite ](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.6-Ic-220/Straight_skeleton_extrusion_2/TestReport_Christo_MSVC2017-Debug-64bits.gz) seems to be independent from the SLS package.

Reply:
What I wrote here is wrong.